rts/Sim/Misc/DefinitionTag.cpp

```diff
@@ -46,36 +46,53 @@ void DefType::AddTagMetaData(const DefTagMetaData* data)
 	}
 
 	tagMetaData[tagMetaDataCnt++] = data;
-}
-
 
-const DefTagMetaData* DefType::GetMetaDataByInternalKey(const string& key)
-{
-	const auto tend = tagMetaData.begin() + tagMetaDataCnt;
-	const auto pred = [&](const DefTagMetaData* md) { return (key == md->GetInternalName()); };
-	const auto iter = std::find_if(tagMetaData.begin(), tend, pred);
+	const std::string internalKey = StringToLower(data->GetInternalName());
+	if (auto it = tagMetaDataByInternalName.find(internalKey);
+			it == tagMetaDataByInternalName.end()) {
+		tagMetaDataByInternalName[internalKey] = data;
+	}
 
-	return ((iter == tend)? nullptr: *iter);
+	if (const auto& externalName = data->GetExternalName(); externalName.IsSet()) {
+		const std::string externalKey = StringToLower(externalName.Get());
+		if (auto it = tagMetaDataByExternalName.find(externalKey);
```

Comment:
Can use `(!tagMetaDataByExternalName.contains(<>))` here and in other places. But best is to assert() or LOG_L() in case of dups.

Reply:
Oh interesting, in that case you can't store both, right? So you can just store the latest with: `tagMetaDataByFallbackName[fallbackKey] = data;`

Reply:
Perfect, t.y.

Reply:
This is a bug though. If multiple tags use the same fallback key then they should all receive the value of that key if the primary key doesn't have a better value. The example above does not show BAR relying on this behaviour in that the linked BAR weapondef wouldn't change if the bug was fixed.